libpimeval/src/pimDevice.cpp

```diff
@@ -137,14 +145,27 @@ pimDevice::isHybridLayoutDevice() const
   return false;
 }
 
-//! @brief  Init pim device, with config file
+//! @brief  Init pim device, with input arguments 
 bool
 pimDevice::init(PimDeviceEnum deviceType, unsigned numRanks, unsigned numBankPerRank, unsigned numSubarrayPerBank, unsigned numRows, unsigned numCols)
 {
   assert(!m_isInit);
   assert(deviceType != PIM_DEVICE_NONE);
 
-  configDevice(deviceType);
+  std::printf("PIM-Info: Trying to read it from envirnment variable %s\n", pimUtils::envVarPimEvalTarget);
+  // Read envirnment variable for the simulation target
+  std::string pimEvalTarget;
+  if (!pimUtils::getEnvVar(pimUtils::envVarPimEvalTarget, pimEvalTarget)) {
+    std::printf("PIM-Info: Could not read environment variable %s\n", pimUtils::envVarPimEvalTarget);
+  }
+  m_simTarget = pimUtils::strToPimDeviceEnum(pimEvalTarget);
+  if (m_simTarget == PIM_DEVICE_UNKNOWN) {
+    std::printf("PIM-Error: Invalid simulation target %s\n", pimEvalTarget.c_str());
```

Comment:
Do we still support make PIM_SIM_TARGET?

Reply:
@deyuan Thanks for noticing this bug. I solved it in new commit.